object_test.go

```diff
@@ -665,64 +665,156 @@ func TestObject_InList(t *testing.T) {
 }
 
 func TestObject_ContainsKey(t *testing.T) {
-	reporter := newMockReporter(t)
-
-	value := NewObject(reporter, map[string]interface{}{"foo": 123, "bar": ""})
-
-	value.ContainsKey("foo")
-	value.chain.assert(t, success)
-	value.chain.clear()
-
-	value.NotContainsKey("foo")
-	value.chain.assert(t, failure)
-	value.chain.clear()
-
-	value.ContainsKey("bar")
-	value.chain.assert(t, success)
-	value.chain.clear()
-
-	value.NotContainsKey("bar")
-	value.chain.assert(t, failure)
-	value.chain.clear()
-
-	value.ContainsKey("BAR")
-	value.chain.assert(t, failure)
-	value.chain.clear()
+	testObj := map[string]interface{}{"foo": 123, "bar": ""}
+	const (
+		contains    = iota
+		notContains = iota
+	)
+
+	cases := []struct {
+		name      string
+		object    map[string]interface{}
+		key       string
+		assertion uint
+		wantEqual chainResult
+	}{
+		{
+			name:      "1. correct key value, contains assertion",
```

Comment:
I'd avoid using numbers in subtest names. It'd require to keep attention that numbers should be unique and correctly ordered when editing tests. That looks like unnecessary work.

Reply:
Same for other tests in this PR.